core/dbt/parser/manifest.py

```diff
@@ -284,6 +284,9 @@ def load(self):
                         exc_info['full_reparse_reason'] = ReparseReason.exception
                         dbt.tracking.track_partial_parser(exc_info)
 
+                    if os.environ.get('DBT_PP_TEST'):
```

Comment:
Is this for testing purposes?

Reply:
Yes. There have been a couple of partial parsing errors that I didn't catch even though there were tests, because of the code that traps the exceptions and does a full parse. So I wanted to make sure we see the exceptions in the test cases.

Reply:
This OS-level trap door makes me think we should split some of this up into better functions or methods. But the tactic definitely works.